debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SnapshotReader.java

```diff
@@ -866,8 +867,31 @@ private void readTableSchema(final AtomicReference<String> sql, final JdbcConnec
         });
     }
 
-    private boolean shouldRecordTableSchema(final MySqlSchema schema, final Filters filters, TableId id) {
-        return !schema.isStoreOnlyMonitoredTablesDdl() || filters.tableFilter().test(id);
+    private boolean shouldRecordTableSchema(final MySqlSchema schema, final Filters filters, TableId id) throws SQLException {
+        return (!schema.isStoreOnlyMonitoredTablesDdl() || filters.tableFilter().test(id)) &&
+                (tableExists(id));
+    }
+
+    /**
+     * Check to see if a table actually exists in the database to handle
+     * cases where <code>SHOW TABLES</code> might erroneously expose 
+     * some internal tables (e.g. <a href="https://issues.redhat.com/browse/DBZ-1939">DBZ-1939</a>).
+     */
+    private boolean tableExists(TableId tableId) throws SQLException {
+        JdbcConnection mysql = connectionContext.jdbc();
+        try {
+            mysql.query("SHOW CREATE TABLE " + quote(tableId), rs -> {
```

Comment:
I think `SHOW ...` are actually the commands that have the issue. Maybe it would have been better to refer to `information_schema` with something like `select 1 from information_schema.tables where table_name = ''tableId";` and check if there are results?

Reply:
So that's interesting: is it guaranteed that `information_schema.tables` doesn't expose that internal table? If indeed it _is_ guaranteed, should we not rather drive the entire determination of tables via that one, instead of SHOW TABLES?

Reply:
OTOH, if `information_schema.tables` _does_ contain that table, it seems doing the suggested try/catch around `SHOW CREATE TABLE` would be the only way to mitigate that situation.

Reply:
Good point, our instances have been fixed (by AWS), but maybe Arik could confirm?

Reply:
Was hoping you guys are right because it would be much nicer to use the `information_schema` instead of parsing an exception , but unfortunately it does look like `show tables` is basically just syntactic sugar around the `information_schema.tables`. Here's what I'm getting:

```
select table_schema,table_name 
from   information_schema.tables
where  table_schema = 'mysql'
and   table_name like 'rds_%';
```

```
mysql	rds_configuration
mysql	rds_global_status_history
mysql	rds_global_status_history_old
mysql	rds_history
mysql	rds_replication_status
mysql	rds_sysinfo
```

Reply:
Hey, finally circling back to this. Sorry for the delay, there was lots of other things going on.

So I've been thinking a bit more about this; wouldn't the following do the trick already:

```java
private boolean shouldRecordTableSchema(final MySqlSchema schema, final Filters filters, TableId id) {
    if (filters.tableFilter().test(id)) {
        return true;
    }
    else {
        return !schema.isStoreOnlyMonitoredTablesDdl();
    }
}
```

WDYT?

Reply:
@gunnarmorling why would that not error when `table.ignore.builtin=false` ?

Reply:
Ah, good point. So how about having a hard-coded list of those RDS tables, which are always going to be ignored then in the filters, independently of that setting? After all, there never should be a case where they should be considered. I'd really prefer to avoid that SHOW TABLE call though due to its potential performance implications.

Reply:
@gunnarmorling  Yea, I like that. That feels a lot more performant and quite frankly less fragile in case that error message changes when the MySQL driver is upgraded. So something like a table-ignore-list filter?

Reply:
Guys we are then getting into the https://github.com/debezium/debezium/pull/1166 scope

Reply:
>  So something like a table-ignore-list filter?

Yes, exactly. That'd always be applied, no matter what's the table.ignore.builtin setting is.

> Guys we are then getting into the #1166 scope

Good point; but are we sure that Replicate_Ignore_Table will expose those tables correctly? If so, yes, then it might be the time to finally wrap up that change, but I'm not so sure given the exisiting inconsistencies around this on Aurora.

Reply:
How does this feel?